src/retrocookie/filter.py

```diff
@@ -33,6 +34,7 @@ def ref(key: str) -> str:
         (value.encode(), ref(key).encode())
         for key, value in context.items()
         if key not in exclude_variables
+        and isinstance(value, str)
         and not (include_variables and key not in include_variables)
```

Comment:
Can we reorder this so the checks for `include_variables` and `exclude_variables` are next to each other?

```suggestion
        and not (include_variables and key not in include_variables)
        and isinstance(value, str)
```